substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/code/CodeInfoEncoder.java

```diff
@@ -393,6 +577,286 @@ public boolean verifyFrameInfo(CodeInfo info) {
+                return (byte) Wrapper.forWrapperType(type).basicTypeChar();
+            } else {
+                throw shouldNotReachHere();
+            }
+        }
+
+        private Annotation[] filterAnnotations(Annotation[] annotations) {
+            List<Annotation> filteredAnnotations = new ArrayList<>();
+            for (Annotation annotation : annotations) {
+                Class<? extends Annotation> annotationClass = annotation.annotationType();
+                if (supportedValue(annotationClass, annotation, null)) {
+                    filteredAnnotations.add(annotation);
+                }
+            }
+            return filteredAnnotations.toArray(new Annotation[0]);
+        }
+
+        private void registerStrings(Annotation[] annotations) {
+            for (Annotation annotation : annotations) {
+                List<String> stringValues = new ArrayList<>();
+                if (supportedValue(annotation.annotationType(), annotation, stringValues)) {
+                    for (String stringValue : stringValues) {
+                        encoders.sourceMethodNames.addObject(stringValue);
+                    }
+                }
+            }
+        }
+
+        @SuppressWarnings("unchecked")
+        private boolean supportedValue(Class<?> type, Object value, List<String> stringValues) {
+            if (type.isAnnotation()) {
+                Annotation annotation = (Annotation) value;
+                if (!encoders.sourceClasses.contains(annotation.annotationType())) {
+                    return false;
+                }
+                AnnotationType annotationType = AnnotationType.getInstance((Class<? extends Annotation>) type);
+                for (Map.Entry<String, Class<?>> entry : annotationType.memberTypes().entrySet()) {
+                    String valueName = entry.getKey();
+                    Class<?> valueType = entry.getValue();
+                    try {
+                        Object annotationValue = annotationType.members().get(valueName).invoke(annotation);
+                        if (!supportedValue(valueType, annotationValue, stringValues)) {
+                            return false;
+                        }
+                    } catch (IllegalAccessException | InvocationTargetException e) {
+                        return false;
+                    }
+                }
+            } else if (type.isArray()) {
+                boolean supported = true;
+                Class<?> componentType = type.getComponentType();
+                if (!componentType.isPrimitive()) {
+                    for (Object val : (Object[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
+                    }
+                } else if (componentType == boolean.class) {
+                    for (boolean val : (boolean[]) value) {
+                        supported &= supportedValue(componentType, val, stringValues);
```

Comment:
Why check all the elements of primitive arrays (for all the array types below)?

Reply:
I indeed don't need to check those, I removed the checks